firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/analytics/AnalyticsConnectorReceiver.java

```diff
@@ -97,48 +98,46 @@ public void unregister() {
   }
 
   @Override
-  public void setCrashlyticsOriginEventListener(@Nullable CrashlyticsOriginEventListener listener) {
-    this.crashOriginEventListener = listener;
-  }
-
-  @Override
-  public @Nullable CrashlyticsOriginEventListener getCrashlyticsOriginEventListener() {
-    return crashOriginEventListener;
+  public void setCrashlyticsAppExceptionEventListener(
+      @Nullable CrashlyticsAppExceptionEventListener listener) {
+    appExceptionEventListener = listener;
   }
 
   @Override
   public void onMessageTriggered(int id, @Nullable Bundle extras) {
-
     Logger.getLogger().d("AnalyticsConnectorReceiver received message: " + id + " " + extras);
 
     if (extras == null) {
       return;
     }
 
-    Bundle params = extras.getBundle(EVENT_PARAMS_KEY);
-    if (params == null) {
-      params = new Bundle();
-    }
+    final String name = extras.getString(EVENT_NAME_KEY);
 
-    final String origin = params.getString(EVENT_ORIGIN_KEY);
-    if (CRASHLYTICS_ORIGIN.equals(origin)) {
-      dispatchCrashlyticsOriginEvent(id, extras);
-    } else {
-      // Drop breadcrumbs for all named events which did not originate from Crashlytics
-      final String name = extras.getString(EVENT_NAME_KEY);
-      if (name != null) {
+    if (name != null) {
+      Bundle params = extras.getBundle(EVENT_PARAMS_KEY);
+      if (params == null) {
+        params = new Bundle();
+      }
+      final String origin = params.getString(EVENT_ORIGIN_KEY);
+      if (CRASHLYTICS_ORIGIN.equals(origin)) {
+        dispatchCrashlyticsOriginEvent(name, params);
+      } else {
+        // Place breadcrumbs for all named events which did not originate from Crashlytics
         dispatchBreadcrumbEvent(name, params);
       }
     }
   }
 
-  private void dispatchCrashlyticsOriginEvent(int id, @Nullable Bundle extras) {
-    if (crashOriginEventListener != null) {
-      crashOriginEventListener.onCrashlyticsOriginEvent(id, extras);
+  private void dispatchCrashlyticsOriginEvent(@NonNull String name, @NonNull Bundle params) {
+    if (appExceptionEventListener != null) {
+      if (APP_EXCEPTION_EVENT_NAME.equals(name)) {
+        Logger.getLogger().d("Received app exception event callback from FA");
```

Comment:
is "FA" how we typically call it in logs? Not GA4F?

Reply:
I think for debugging output it doesn't matter much (and I may have initially been the one to put "FA" in there), but the Google Analytics SDK team prefers "Google Analytics for Firebase SDK".

Reply:
Yeah, Marc originally wrote this log statement. I thought it was intended to map to the Firebase Analytics log tag, which is "FA". Want me to update?